payload/payload.cc

```diff
@@ -10,68 +10,73 @@ namespace sorbet::payload {
 
 void createInitialGlobalState(unique_ptr<core::GlobalState> &gs, const realmain::options::Options &options,
                               const unique_ptr<const OwnedKeyValueStore> &kvstore) {
-    if (kvstore) {
-        ENFORCE(!options.noStdlib, "--no-stdlib and --cache-dir are incompatible");
-        auto maybeGsBytes = kvstore->read(core::serialize::Serializer::GLOBAL_STATE_KEY);
-        if (maybeGsBytes.data != nullptr) {
-            Timer timeit(gs->tracer(), "read_global_state.kvstore");
-            core::serialize::Serializer::loadGlobalState(*gs, maybeGsBytes.data);
-            for (unsigned int i = 1; i < gs->filesUsed(); i++) {
-                core::FileRef fref(i);
-                if (fref.dataAllowingUnsafe(*gs).sourceType == core::File::Type::Normal) {
-                    gs = core::GlobalState::markFileAsTombStone(move(gs), fref);
-                }
-            }
-            return;
-        }
-    }
     if (options.noStdlib) {
         gs->initEmpty();
-        return;
-    }
-
-    const uint8_t *const nameTablePayload = getNameTablePayload;
-    if (nameTablePayload == nullptr) {
-        Timer timeit(gs->tracer(), "read_global_state.source");
-        sorbet::rbi::populateRBIsInto(gs);
     } else {
-        Timer timeit(gs->tracer(), "read_global_state.binary");
-        core::serialize::Serializer::loadGlobalState(*gs, nameTablePayload);
-    }
-    ENFORCE(gs->utf8NamesUsed() < core::GlobalState::PAYLOAD_MAX_UTF8_NAME_COUNT,
+        const uint8_t *const nameTablePayload = getNameTablePayload;
```

Comment:
It's probably time to rename this variable. "name table" is what Dmitry used to call GlobalState. Now that we explicitly have a concept of "binary-serialized name table," continuing to have this name is particularly confusing.

Reply:
Hah, I was wondering about the origin of this! What do you think about handling that in a follow-up?

Reply:
wfm